sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java

```diff
@@ -100,12 +100,16 @@
      * @throws RuntimeException If the directory is using a malformed URL.
      */
     public URL getDirectoryUrl() {
+        String directoryURLString = String.format("%s/%s/%s", azureFileStorageClient.getUrl(),
```

Comment:
Do we split out the share and directory path in the URL? If not we don't need this `String.format`.

Reply:
The format is like
`https://sima.file.core.windows.net/sharename/directoryPath`